lib/concurrent/edge/future.rb

```diff
@@ -925,8 +927,11 @@ def complete_with(new_state, raise_on_reassign = true)
       # @return [Future]
       def evaluate_to(*args, block)
         complete_with Future::Success.new(block.call(*args))
-      rescue => error
+      rescue StandardError => error
         complete_with Future::Failed.new(error)
+      rescue Exception => error
+        complete_with Future::Failed.new(error)
+        log(ERROR, 'Edge::Future', error)
```

Comment:
Nice catch, it would break the documented behaviour, it would break `CompletableFuture#evaluate_to`. Fixing and merging.